asynq/async_task.py

```diff
@@ -122,10 +122,10 @@ def _continue(self):
             try:
                 self._accept_yield_result(self._continue_on_generator(value, error))
             except StopIteration as error:  # Most frequent, so it's the first one
-                if hasattr(error, 'value'):
+                try:
                     # We're on a Python version that supports adding a value to StopIteration
                     self._queue_exit(error.value)
-                else:
```

Comment:
Strictly the try-except should only cover the call to error.value. If _queue_exit throws an AttributeError due to a bug, we shouldn't catch the error. So we should do something like `try: return_value = error.value except AttributeError: return_value = None self._queue_exit(return_value)`